src/libstd/ffi/os_str.rs

```diff
@@ -277,6 +277,18 @@ impl OsStr {
         self.to_bytes().and_then(|b| CString::new(b).ok())
     }
 
+    /// Returns true if the `other` is a prefix of the `OsStr`.
+    #[unstable(feature = "os_str_compare", reason = "recently added")]
+    pub fn starts_with<S: AsRef<OsStr>>(&self, other: S) -> bool {
+        self.bytes().starts_with(other.as_ref().bytes())
```

Comment:
This is correct for plain bytes and UTF-8, but what about other encodings? If `OsStr` is WTF-8, does that allow false positives (say for example that a single byte encoded sequence for a codepoint exists that's a prefix of a multibyte sequence. UTF-8 doesn't have this).

Reply:
* Just a note for people from the future: No this is **not fine** because if `self` contains a non-BMP code point (4-byte sequence), and `other` contains just a high-surrogate (3-byte sequence), this implementation will always return `false`, contradicting the result when given two WTF-16 sequences.